sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -576,6 +578,27 @@ trait Column extends DataFrame {
   override def as(alias: Symbol): Column = exprToColumn(Alias(expr, alias.name)())
 
   /**
+   * (Scala-specific) Explodes the column to zero or more rows by the provided function.
+   * {{{
+   *   val df = Seq(Tuple1("a b c"), Tuple1("d e")).toDataFrame("words")
+   *   val col = df("words")
+   *   col.explode("word"){words: String => words.split(" ")}
+   * }}}
+   */
+  def explode[A, B : TypeTag](
+      outputColumn: String)(
```

Comment:
Okay, that is better. It should not break overloading because its signature is different to DataFrame's.